app/plugins.go

```diff
@@ -365,29 +410,105 @@ func (a *App) RemovePlugin(id string) *model.AppError {
 		return model.NewAppError("RemovePlugin", "app.plugin.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	plugins := a.PluginEnv.ActivePlugins()
-	manifest := &model.Manifest{}
+	plugins, err := a.PluginEnv.Plugins()
+	if err != nil {
+		return model.NewAppError("RemovePlugin", "app.plugin.deactivate.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+
+	var manifest *model.Manifest
 	for _, p := range plugins {
 		if p.Manifest.Id == id {
 			manifest = p.Manifest
 			break
 		}
 	}
 
-	err := a.PluginEnv.DeactivatePlugin(id)
-	if err != nil {
-		return model.NewAppError("RemovePlugin", "app.plugin.deactivate.app_error", nil, err.Error(), http.StatusBadRequest)
+	if manifest == nil {
+		return model.NewAppError("RemovePlugin", "app.plugin.not_installed.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	if a.PluginEnv.IsPluginActive(id) {
+		err := a.PluginEnv.DeactivatePlugin(id)
+		if err != nil {
+			return model.NewAppError("RemovePlugin", "app.plugin.deactivate.app_error", nil, err.Error(), http.StatusBadRequest)
+		}
+
+		if manifest.HasClient() {
+			message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_DEACTIVATED, "", "", "", nil)
+			message.Add("manifest", manifest.ClientManifest())
+			a.Publish(message)
+		}
 	}
 
 	err = os.RemoveAll(filepath.Join(a.PluginEnv.SearchPath(), id))
 	if err != nil {
 		return model.NewAppError("RemovePlugin", "app.plugin.remove.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	if manifest.HasClient() {
-		message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_DEACTIVATED, "", "", "", nil)
-		message.Add("manifest", manifest.ClientManifest())
-		a.Publish(message)
+	return nil
+}
+
+// EnablePlugin will set the config for an installed plugin to enabled, triggering activation if inactive.
+func (a *App) EnablePlugin(id string) *model.AppError {
+	if a.PluginEnv == nil || !*a.Config().PluginSettings.Enable {
+		return model.NewAppError("RemovePlugin", "app.plugin.disabled.app_error", nil, "", http.StatusNotImplemented)
+	}
+
+	plugins, err := a.PluginEnv.Plugins()
+	if err != nil {
+		return model.NewAppError("EnablePlugin", "app.plugin.config.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	var manifest *model.Manifest
+	for _, p := range plugins {
+		if p.Manifest.Id == id {
+			manifest = p.Manifest
+			break
+		}
+	}
+
+	if manifest == nil {
+		return model.NewAppError("EnablePlugin", "app.plugin.not_installed.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	cfg := a.Config()
+	cfg.PluginSettings.PluginStates[id] = &model.PluginState{Enable: true}
+
+	if err := a.SaveConfig(cfg, true); err != nil {
```

Comment:
This will trigger the config listener, which will activate/deactivate plugins as necessary